pkg/reconciler/pipeline/controller.go

```diff
@@ -76,11 +77,16 @@ func (r *pipelineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	log := r.logger.With("namespace", pl.Namespace).With("pipeline", pl.Name)
 	plCopy := pl.DeepCopy()
 	ctx = logging.WithLogger(ctx, log)
+	if err := plCopy.Spec.Templates.UpdateWithDefaultsFrom(r.templates); err != nil {
+		r.logger.Errorw("Unable to create merged pipeline templates", err)
+		return ctrl.Result{}, err
+	}
 
 	result, reconcileErr := r.reconcile(ctx, plCopy)
 	if reconcileErr != nil {
 		log.Errorw("Reconcile error", zap.Error(reconcileErr))
 	}
+	plCopy.Spec.Templates = pl.Spec.Templates.DeepCopy()
```

Comment:
templates do not get changed in `reconcile()`, but they do get changed from `pl` immediately after plCopy is created in Reconcile().

Mostly I was trying to find a good way to do the cluster-wide + pipeline-wide templates merge once & not need to pass down the merged templates through the func calls.  However maybe each `job`, `daemon`, `vertex` component can handle their own merge to avoid changing it... will look at that.